test/unit/communication/test_comm_sender.cpp

```diff
@@ -64,9 +63,8 @@ TEST(Bcast, Polling) {
   double val = (comm.rank() == root_rank) ? 4.2 : 1.2;
   std::vector<double> buf(static_cast<std::size_t>(size), val);
 
-  sync_wait(when_all(just(buf.data()), pika::make_ready_future<int>(size), just(dtype, root_rank, comm),
-                     pika::make_ready_future<void>()) |
-            transformMPI(MPI_Ibcast) | then([](int e) { DLAF_MPI_CHECK_ERROR(e); }));
+  sync_wait(when_all(just(buf.data(), size, dtype, root_rank, comm)) | transformMPI(MPI_Ibcast) |
```

Comment:
Is the `when_all` needed?

Reply:
Nope, good catch. Removed in https://github.com/eth-cscs/DLA-Future/pull/908/commits/bda63508ee2127713df21ce0888553f054f414e6.